### PR TITLE
Move current image info to device-registry.

### PR DESCRIPTION
**Pull request by houcros**
Based on #64.

### CODE REVIEW

**Comment by jerrytrieu** [comment]
@houcros can you add here all the APIs in Director that deal with current images?

**Comment by kmanikandan-kmk** [comment]
Does moving ECU image info to Device Registry impact any of the recommendations made from Uptane for Director? Because, moving ECU info to Device Registry means Director has to rely on Device Registry for the information about Devices, ECUs and images.

https://docs.google.com/document/d/1wjg3hl0iDLNh7jIRaHl3IXhwm0ssOtDje5NemyTBcaw/edit#heading=h.x97q3jyvlka9

As per section 6.1 ([Directing installation of images on vehicles](https://docs.google.com/document/d/1wjg3hl0iDLNh7jIRaHl3IXhwm0ssOtDje5NemyTBcaw/edit#heading=h.6j0ugafdu56g)), Director is expected to have enough information to identify the images to be installed on devices.

**Comment by houcros** [comment]
@jerrytrieu I've added the endpoints and tried to improved them a bit:

```
GET /api/v1/admin/images/affected               => GET /api/v1/images/affected
POST /api/v1/admin/images/installed_count => GET /api/v1/images/installed_count
GET /api/v1/admin/devices/<uuid>                => GET /api/v1/devices/<uuid>/images
GET /api/v1/admin/devices/<uuid>/images  => Removed because it was returning a subset of the info in the next endpoint.
```

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/db/MigrateEcuInfoFromDirector.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  private val _log = LoggerFactory.getLogger(this.getClass)
+
+  def saveEcu(ecu: Ecu): Future[EcuIdentifier] = {
+    _log.info(s"Migrating ECU from director: $ecu.")
+    db.run(EcuRepository.ecus.insertOrUpdate(ecu).map(_ => ecu.ecuId))
+  }
+
+  def run: Future[Done] = {
+    val source = db.stream(devices.map(d => (d.namespace, d.uuid)).result)
+    Source
```
Comment (simao) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/db/MigrateEcuInfoFromDirector.scala]:
```scala
Source
      .fromPublisher(source)
      .mapAsyncUnordered(1) { case (ns, did) => director.getEcuInfo(ns, did).map(_.map((ns, did, _))) }
      .mapConcat(_.toList)
      .mapAsyncUnordered(4) { case (ns, did, ecuInfo) =>
        for {
          pubKey <- director.getPublicKey(ns, did, ecuInfo.id)
          ecuId <- saveEcu(Ecu(did, ecuInfo.id, ecuInfo.hardwareId, ecuInfo.primary, pubKey))
        } yield ecuId
      }
      .runWith {
        Sink.foreach(ecuId => _log.info(s"Migrated ECU $ecuId"))
      }
```

Similar for the other migration

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/client/DirectorClient.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import eu.timepit.refined.api.Refined
+import io.circe.Decoder
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final case class EcuInfo(id: EcuIdentifier, hardwareId: HardwareIdentifier, primary: Boolean)
+
+trait DirectorClient {
```
Comment (simao) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/client/DirectorClient.scala]: I thought this was going to director ?

Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/client/DirectorClient.scala]: Eventually yes. But when I started this PR we didn't have that in mind yet. I might try to squeeze that in if I manage to create the other client on time.

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/db/MigrateEcuInfoFromDirector.scala (hunk 1 above)
Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/db/MigrateEcuInfoFromDirector.scala]: Updated.

**Comment by houcros** [comment]
Same reason than #64.

**Comment by houcros** [comment]
Same reason than #64.

**Pull request closed without merging**